### PR TITLE
rel to #12409: strip permissions from searchactivity intent before giving it back

### PR DESCRIPTION
**Pull request by eddiemuc**
rel to #12409: strip permissions from searchactivity intent before giving it back

### CODE REVIEW

File: main/src/cgeo/geocaching/SearchActivity.java [hunk 1]
```diff
@@ -123,6 +133,17 @@ public final void onCreate(final Bundle savedInstanceState) {
         init();
     }
 
+    private static void removeFlags(final Intent intent, final int flags) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/SearchActivity.java:137]: Isn't this IF statement doubled with the calling part above?

Comment (eddiemuc) [reply main/src/cgeo/geocaching/SearchActivity.java:137]: Forgot to delete unused method, will remove it (codacy complains too...)

**Comment by eddiemuc** [comment]
BTW: if this PR does NOT remove the warning, then I propose simply to not send the intent data back to the caller in the assumed problematic line

**Comment by Lineflyer** [comment]
FYA:
I just merged branches!
If this shall be part of next beta you should please re-target it to `release` now!

**Comment by eddiemuc** [comment]
> FYA: I just merged branches! If this shall be part of next beta you should please re-target it to `release` now!

done

**Comment by Lineflyer** [comment]
Can someone provide me some details about how to potentially test if this has any side effects?
When is this code line / function (Searchactivity intent) used? What do I have to do?

**Comment by eddiemuc** [comment]
> Can someone provide me some details about how to potentially test if this has any side effects? When is this code line / function (Searchactivity intent) used? What do I have to do?

Not exactly. Somehow you have to start a search from another app which is handled by c:geo but fails. 

In this special case it might be wise to first check whether the fix really solves the reported security issue and THEN test it afterwards - since the former seems to  be easier to achieve, and if the fix does not help with the security report then we have to abandon it anyway.

**Comment by Lineflyer** [comment]
> Not exactly. Somehow you have to start a search from another app which is handled by c:geo but fails.

Do you mean "search" like e.g. a nearby search?

Could the problem itself be caused by the relatively new feature to trigger a nearby search from shared coordinates from another app?
E.g. Open Adventure Lab app, select a stage, use navigation icon --> Select "Search Caches Nearby with c:geo" as target

**Comment by eddiemuc** [comment]
> Can someone provide me some details about how to potentially test if this has any side effects? When is this code line / function (Searchactivity intent) used? What do I have to do?



> > Not exactly. Somehow you have to start a search from another app which is handled by c:geo but fails.
> 
> Do you mean "search" like e.g. a nearby search?

No. As far as I can see from code, this search tries to find and extract a geocode from the URL passed with the intent, and then it searches for a cache with this geocode.

(There's a side path in code which also is doing a keyword search, but intent has to have rather c:geo-specific fields for this path to be chosen)

I have to admit that I don't know this code part very well.

As said before: I recommend first to try whether this code version will fix the security warning of Google before going into further testing.

**Comment by Lineflyer** [comment]
> As said before: I recommend first to try whether this code version will fix the security warning of Google before going into further testing.

Then probably its better to merge to `master` only for now. Afterwards I can push nightlies to Googles test engine again to see if something changes.

OK?

**Comment by eddiemuc** [comment]
Ok, will merge to master

**Pull request merged**